src/ThemeProvider.tsx

```diff
@@ -101,10 +101,6 @@ export const ThemeProvider: React.FC<ThemeProviderProps> = ({children, ...props}
     setColorMode(props.colorMode ?? fallbackColorMode ?? defaultColorMode)
   }, [props.colorMode, fallbackColorMode])
 
-  React.useEffect(() => {
-    setColorMode(resolvedColorMode)
-  }, [resolvedColorMode])
-
```

Comment:
I split this effect in https://github.com/primer/react/pull/1936, now this effect was just syncing state to colorMode 🤔  which is a bug

Reply:
I don't think I understand this bug. Everything seems to be working as expected 🤔

Reply:
The bug was that changes in system settings did not change the resolved theme in ThemeProvider

repro:

https://user-images.githubusercontent.com/1863771/165322367-37ed0ef5-497c-4473-9c21-cb4718d3cd56.mov